### PR TITLE
prevent bad kwargs from being forwarded to precompile

### PR DESCRIPTION
**Pull request by KristofferC**
Should fix the breakage on nightly. https://travis-ci.com/github/JuliaLang/PackageCompiler.jl/jobs/402537410#L551

~Going to add a test.~

### CODE REVIEW

File: src/API.jl [hunk 1]
```diff
@@ -907,16 +907,17 @@ function _is_stale(paths::Vector{String}, sourcepath::String)
...
+function _auto_precompile(ctx::Context)
     if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "1")) == 1
-        Pkg.precompile(ctx; internal_call=true, kwargs...)
+        Pkg.precompile(ctx; internal_call=true)
     end
 end
 
 precompile(; kwargs...) = precompile(Context(); kwargs...)
-function precompile(ctx::Context; internal_call::Bool=false, io::IO=stderr)
+function precompile(ctx::Context; internal_call::Bool=false)
```
Comment (fredrikekre) [inline src/API.jl:917]: Still needs `io` though? But it shoule be consumed by the context

Comment (KristofferC) [reply src/API.jl]: Yeah, tweaked it slightly.

File: src/API.jl [hunk 2]
```diff
@@ -907,16 +907,18 @@ function _is_stale(paths::Vector{String}, sourcepath::String)
...
+function _auto_precompile(ctx::Context)
     if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "1")) == 1
-        Pkg.precompile(ctx; internal_call=true, kwargs...)
+        Pkg.precompile(ctx; internal_call=true)
     end
 end
 
 precompile(; kwargs...) = precompile(Context(); kwargs...)
-function precompile(ctx::Context; internal_call::Bool=false, io::IO=stderr)
+function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
```
Comment (fredrikekre) [inline src/API.jl:917]:
```suggestion
function precompile(ctx::Context; io::IO=stderr, internal_call::Bool=false, kwargs...)
```

File: src/API.jl [hunk 3]
```diff
@@ -907,16 +907,18 @@ function _is_stale(paths::Vector{String}, sourcepath::String)
...
+    Context!(ctx; kwargs...)
```
Comment (fredrikekre) [inline src/API.jl:918]:
```suggestion
    Context!(ctx; io=io, kwargs...)
```

**Comment by DilumAluthge** [comment]
Travis failures look like the usual spurious download/network failures.

**Comment by DilumAluthge** [comment]
Thanks @KristofferC and @fredrikekre for getting this PR so quickly!

**Comment by KristofferC** [comment]
> Apologies for causing this... Thanks for the quick fix!

No problem. The flow of keyword arguments is really quite awkward right now, and I'm the one to blame :)

**Comment by DilumAluthge** [comment]
Thanks for the quick turnaround! Can you also bump the commit of Pkg in Julia master?

**Comment by KristofferC** [comment]
I'll just try sneak in https://github.com/JuliaLang/Pkg.jl/pull/2143/files

**Pull request merged**